src/index.tsx

```diff
@@ -1501,36 +1801,166 @@ export type UseInputField<Value> = {
   state: UseFieldState<Value>;
 };
 
-export type UseInputFieldProps<Value> = {
+export type UseInputFieldProps<Type extends React.HTMLInputTypeAttribute> = {
   /**
    * The name of the field if there is one
    */
   name: string | undefined;
   /**
    * The value of the field
    */
-  value: Value;
+  value: Type extends DateType
+    ? string
+    : Type extends NumberType
+    ? number
+    : Type extends FileList
```

Comment:
Here it should be `extends FIleType`

as currently I have  `value: string`
![image](https://user-images.githubusercontent.com/38855190/217803667-83348c9b-fb72-46c4-babb-7496053684c5.png)

Still I don't use the value, as the file input is uncontrolled, but still.

Reply:
So the type likely should be:
```
 value: Type extends DateType
    ? string
    : Type extends NumberType
    ? number
    : Type extends FileType // here 
    ? FileList // and here
    : string;
````